how-to-build.md

```diff
@@ -46,110 +42,64 @@ On OS X, Clang is used as the C/C++ compiler by default. Clang is installed
 as part of Xcode. Xcode 7.0 or newer is required, and can be installed via
 the Mac App Store.
 
-Configure, build & test
--------------------------
+## Configure, build & test
 
 Run the following commands to configure, build and test core:
 
-    sh build.sh config
-    sh build.sh build
-    sh build.sh test
-
-
-Building for Android
---------------------
-
-Building for Android required the NDK of version >= r10d installed.
-
-It can be built using the following command:
-
-    sh build.sh build-android
-
-Building for iOS, watchOS and tvOS
-----------------------------------
-
-Realm core may be built for iOS, watchOS and tvOS from an OS X machine.
-See the OS X section of [Prerequisites](#prerequisites) above for information
-about the version of Xcode that is required.
-
-To build for iOS:
-
-    sh build.sh build-ios
-
-This produces the following files and directories:
+    mkdir build-dir # create a build folder
+    cd build-dir
+    cmake ..
+    cmake --build
+    ctest
 
-    ios-lib/include/
-    ios-lib/librealm-ios.a
-    ios-lib/librealm-ios-dbg.a
-    ios-lib/realm-config
-    ios-lib/realm-config-dbg
+## Building for Android, iOS, watchOS and tvOS
 
-The `include` directory holds a copy of the header files, which are
-identical to the ones installed by `sh build.sh install`. There are
-two versions of the static library, one that is compiled with
-optimization, and one that is compiled for debugging. Each one
-contains code compiled for both iOS devices and the iOS simulator.
-Each one also comes with a `config` program that can be used to
-enquire about required compiler and linker flags.
+Building for Android required the NDK r10e installed.
 
-To build for watchOS:
+These targets can be built using the build.sh command:
 
-    sh build.sh build-watchos
+    ./build.sh -o android -a armeabi-v7a -t Release -v <X.Y.Z>
 
-The output is placed in watchos-lib, and is structured the same as
-is described for iOS above.
+The command shows the available options simply with:
 
-To build for tvOS:
+    ./build.sh
 
-    sh build.sh build-tvos
+These commands produce a tarball containing the realm static library
+and its include files.
```

Comment:
Have we started on the work in sync and the binding to adapt to this new structure to be sure that they can usefully consume what this PR is producing? That seems like something we need to be sure of before we can merge this PR.

Another side of this that I brought up on Slack was what workflow we expect developers to use when testing core changes against the Cocoa binding. Currently `sh build.sh build-cocoa` produces something similar to the release tarball, but also deposits the files in the appropriate place for the Cocoa binding to build against it. When I'm testing changes I currently do the following to build both core and sync and have Cocoa use my built versions:
```
cd realm-core && REALM_COCOA_PLATFORMS="osx" sh build.sh build-cocoa ../realm-cocoa
cd realm-sync && REALM_COCOA_PLATFORMS="osx" sh build.sh build-cocoa ../realm-cocoa
```

I can tweak the `REALM_COCOA_PLATFORMS` environment variable depending on which platforms I need to test, or omit it entirely if I want to build the full set of platforms.

It's not at all clear to me how to do the equivalent thing on this branch.